app/src/main/java/com/globant/harrypotterapp/fragment/HouseDetailFragment.kt

```diff
@@ -51,20 +52,20 @@ class HouseDetailFragment : Fragment() {
                 RAVENCLAW -> binding.houseDetailFragmentImage.setImageResource(R.drawable.ravenclaw)
                 SLYTHERIN -> binding.houseDetailFragmentImage.setImageResource(R.drawable.slytherin)
                 HUFFLEPUFF -> binding.houseDetailFragmentImage.setImageResource(R.drawable.hufflepuff)
-                else -> showErrorMessage()
             }
         }
     }
 
-    private fun showErrorMessage() {
-        binding.houseDetailFragmentLoader.visibility = View.GONE
-        binding.fragmentHouseDetailRoot.background = context?.getDrawable(R.drawable.hogwarts)
-        Toast.makeText(this.context, getString(R.string.house_detail_fragment_error_message), Toast.LENGTH_SHORT).show()
+    private fun showHouseDetailError(message: String?) {
+        with(binding) {
+            houseDetailFragmentLoader.visibility = View.GONE
+            houseDetailFragmentError.visibility = View.VISIBLE
+            houseDetailFragmentError.text = getString(R.string.error_message_text, message, HOUSE_DETAILS)
+        }
     }
 
     companion object {
-        private const val ZERO = 0
-        private const val GRYFFINDOR = "Gryffindor"
+        private const val HOUSE_DETAILS = "House details"
         private const val RAVENCLAW = "Ravenclaw"
```

Comment:
same, put the houses names in the constant utils

Reply:
Done!